xcube/webapi/auth.py

```diff
@@ -199,38 +234,60 @@ def get_id_token(self, require_auth: bool = False) \
                     "e": key["e"]
                 }
                 break
+        if rsa_key is None:
+            raise ServiceAuthError(
+                "Invalid header",
+                log_message="Unable to find appropriate key."
+            )
 
-        if rsa_key:
-            try:
-                id_token = jwt.decode(
-                    access_token,
-                    # TODO: this is stupid: we convert rsa_key to
-                    #  JWT JSON only to produce the public key JSON string
-                    RSAAlgorithm.from_jwk(json.dumps(rsa_key)),
-                    algorithms=auth_config.algorithms,
-                    audience=auth_config.audience,
-                    issuer=auth_config.issuer
-                )
-            except jwt.ExpiredSignatureError:
-                raise ServiceAuthError(
-                    "Token expired",
-                    log_message="Token is expired"
-                )
-            except jwt.InvalidTokenError:
-                raise ServiceAuthError(
-                    "Invalid claims",
-                    log_message="Incorrect claims,"
-                                " please check the audience and issuer"
-                )
-            except Exception:
+        # Now we are ready to decode the access token
+        try:
+            id_token = jwt.decode(
+                access_token,
+                RSAAlgorithm.from_jwk(rsa_key),
+                algorithms=auth_config.algorithms,
+                audience=auth_config.audience,
+                issuer=auth_config.issuer
+            )
+        except jwt.PyJWTError as e:
+            msg = f"Failed to decode access token: {e}"
+            raise ServiceAuthError(msg, log_message=msg) from e
+
+        return id_token
+
+    def get_access_token(self, require_auth: bool = False) -> Optional[str]:
+        """
+        Obtain the request's access token from the Authorization Header.
+        """
+        # noinspection PyUnresolvedReferences
+        auth = self.request.headers.get("Authorization", None)
+        if not auth:
+            if require_auth:
                 raise ServiceAuthError(
-                    "Invalid header",
-                    log_message="Unable to parse authentication token."
+                    "Authorization header missing",
+                    log_message="Authorization header is expected"
                 )
-            return id_token
+            return None
+
+        parts = auth.split()
 
-        raise ServiceAuthError("Invalid header",
-                               log_message="Unable to find appropriate key")
+        if parts[0].lower() != "bearer":
+            raise ServiceAuthError(
+                "Invalid header",
+                log_message='Authorization header must start with "Bearer"'
```

Comment:
Because afaik, HTTP request header names are case in-sensitive, but as a common practice, the are written with an upper case first letter. (Do you want me to enumerate all possible literal cases of "Bearer"?)